src/main/java/org/opensearch/ad/rest/RestCreateDetectorAction.java

```diff
@@ -66,8 +61,8 @@ public RestCreateDetectorAction(ExtensionsRunner runner, AnomalyDetectorExtensio
     }
 
     @Override
-    public List<Route> routes() {
-        return List.of(new Route(POST, "/detectors"));
+    protected List<RouteHandler> routeHandlers() {
+        return List.of(new RouteHandler(POST, "/detectors", (r) -> handlePostRequest(r)));
```

Comment:
@owaiskazi19 FYI, I tried to use the same shortened format as the `RestHelloAction` but the internal use of `client` and `xContentRegistry` inside a `Function` falls afoul of "must be final" (even though they are defined final!)

So I had to keep it as a standard method and do the lambda version of the `Function` inline here.